examples/plot_bids_analysis.py

```diff
@@ -0,0 +1,114 @@
+"""
+BIDS dataset first and second level analysis
+============================================
+
+Full step-by-step example of fitting a GLM to perform a first and second level
+analysis in a BIDS dataset and visualizing the results. Details about the BIDS
+standard can be consulted at https://openfmri.org/data-organization/
+
+More specifically:
+
+1. Download an fMRI BIDS dataset with two language conditions to contrast.
+2. We extract automatically from the BIDS dataset first level model objects
+3. We fit a second level model on the fitted first level models. Notice that
+   in this case the preprocessed bold images were already normalized to the
+   same MNI space.
+
+Author : Martin Perez-Guevara: 2016
+"""
+import os
+from nilearn import plotting
+from scipy.stats import norm
+import matplotlib.pyplot as plt
+
+from nistats.datasets import fetch_bids_langloc_dataset
+from nistats.first_level_model import first_level_models_from_bids
+from nistats.second_level_model import SecondLevelModel
+
+#############################################################################
+# Fetch example BIDS dataset
+# --------------------------
+# We download a partial example BIDS dataset. It contains only the necessary
+# information to run an statistical analysis using Nistats. The raw data
+# subject folders only contain bold.json and events.tsv files, while the
+# derivatives folder with preprocessed files contain preproc.nii and
+# confounds.tsv files.
+data_dir, _ = fetch_bids_langloc_dataset()
+
+#############################################################################
+# Obtain automatically FirstLevelModel objects and fit arguments
+# --------------------------------------------------------------
+# From the dataset directory we obtain automatically FirstLevelModel objects
+# with their subject_id filled from the BIDS dataset. Moreover we obtain
+# for each model a dictionary with run_imgs, events and confounder regressors
+# since in this case a confounds.tsv file is available in the BIDS dataset.
+# To get the first level models we only have to specify the dataset directory
+# and the task_id as specified in the file names.
+task_id = 'languagelocalizer'
```

Comment:
not if we follow BIDS specification. The specification says: "Each task has a unique label consisting of letters and/or numbers (other characters including spaces and underscores are not allowed)."